internal/db/db.go

```diff
@@ -235,6 +235,11 @@ func (db *DB) waitUpgrade(bootstrap bool, ext extensions.Extensions) error {
 
 // Transaction handles performing a transaction on the dqlite database.
 func (db *DB) Transaction(outerCtx context.Context, f func(context.Context, *sql.Tx) error) error {
+	status := db.Status()
+	if status != StatusWaiting && status != StatusReady {
```

Comment:
When the DB is in waiting state, why is it fine to continue with the transaction?

Reply:
A waiting state means the database is open, but is waiting for the schema update to appear on other systems, which means some functionality is restricted.

The database is technically fully functional at this time, so we should be able to query it on endpoints where `AllowedBeforeInit=true`.

This is useful for querying the current upgrade status, like in #162

Additionally, it allows for the downstream project to still start the underlying services like OVN and Ceph while awaiting a dqlite schema change.